### PR TITLE
Smtp integration

### PR DESCRIPTION
**Pull request by Thurfjell**
### Ticket:

- [15227](https://dev.azure.com/OGCIO-Digital-Services/Digital%20Services%20Programme/_sprints/taskboard/Life%20Events/Digital%20Services%20Programme/Mar%202024/Mar%2013%20-%2026?System.Tags=Life%20Events%2Cmessaging&System.AssignedTo=ludwig.thurfjell%40nearform.com&workitem=15227)

### Description

Add nodemailer package to message package.
Add a smtp crud form for admin message settings.
Automatically create a ethereal.email account if none exists for a "dev" account, this service intercepts the smtp call and just show the message (read more at https://ethereal.email ). Good for dev debugging.
Automatically (for now) uses the temporary mail provider template when choosing email as transport for messages.

To debug/test, send a message and keep emails selected, proceed as normal.
This should send the email together with the normal message. 
The username/password will be available at `admin/settings/emails ` in messaging app. 

## Type

- [ ] **Dependency upgrade**
- [ ] **Bug fix**
- [x] **New feature**
- [x] **Dev change**

### Checklist:

- [ ] I have added tests that prove my fix is effective or that my feature works

### Screenshots:

N/A

### CODE REVIEW

File: packages/messages/.env.sample [hunk 1]
```diff
@@ -3,4 +3,5 @@ POSTGRES_USER=postgres
 POSTGRES_PASSWORD=postgres
 POSTGRES_HOST=localhost
 POSTGRES_PORT=5432
-POSTGRES_DB_NAME_MESSAGING=messaging
\ No newline at end of file
+POSTGRES_DB_NAME_MESSAGING=messaging
+SMTP_PORT=587
```
Comment (Thurfjell) [inline packages/messages/.env.sample:7]: @williamwd 
I'm not sure this is how we wanna go about these things. Any suggestions?

File: packages/messages/index.ts [hunk 2]
```diff
@@ -27,6 +28,119 @@ type MessageState = {
...
+        host,
+        port: Number(process.env.SMTP_PORT) || 0,
+        // secure: false,
+        auth: {
+          user: username,
+          pass: password,
+        },
+      });
+
+      console.log({ recipients });
```
Comment (GiuliaTeggi) [inline packages/messages/index.ts:129]:
```suggestion
```

File: packages/messages/migrations/20240403082727-nodemailer.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+exports.up = function (db) {
+  var filePath = path.join(
+    __dirname,
+    "sqls",
+    "20240403082727-nodemailer-up.sql",
+  );
+  return new Promise(function (resolve, reject) {
+    fs.readFile(filePath, { encoding: "utf-8" }, function (err, data) {
+      if (err) return reject(err);
+      console.log("received data: " + data);
```
Comment (GiuliaTeggi) [inline packages/messages/migrations/20240403082727-nodemailer.js:30]: Do you need to keep this log?

File: packages/messages/migrations/20240403082727-nodemailer.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+exports.down = function (db) {
+  var filePath = path.join(
+    __dirname,
+    "sqls",
+    "20240403082727-nodemailer-down.sql",
+  );
+  return new Promise(function (resolve, reject) {
+    fs.readFile(filePath, { encoding: "utf-8" }, function (err, data) {
+      if (err) return reject(err);
+      console.log("received data: " + data);
```
Comment (GiuliaTeggi) [inline packages/messages/migrations/20240403082727-nodemailer.js:48]: And this one?

**Review by GiuliaTeggi — APPROVED** [review]

File: packages/messages/migrations/20240403082727-nodemailer.js (hunk 3 above)
Comment (Thurfjell) [reply packages/messages/migrations/20240403082727-nodemailer.js:30]: No idea, they are auto generated.

**Pull request merged**